src/gill/TrouserPattern.java

```diff
@@ -154,470 +154,146 @@ protected void createBlocks()
-        }
-        else if (r_CrotchZ > 0)
-        {
-            stepThirteenY = ((x_WaistBkZ + r_CrotchZ) - (2.5 * Arb_CrotchReduction) + ((v_SeatBkZ - x_WaistBkZ) * 2.0));
-        }
-        else if (r_CrotchZ == 0)
-        {
-            stepThirteenY = ((x_WaistBkZ) - (2.5 * Arb_CrotchReduction) + ((v_SeatBkZ - x_WaistBkZ) * 2.0));
-        }
-
-        fullBlock.addKeypoint(new Vector2D(-k_BodyRise, stepThirteenY));
+        // Create component representing half back of skirt folded in half.
+        Block fullBlock = new Block(userName + "_Gill_Trouser_Block");
+        blocks.add(fullBlock);
 
-        // Adding Step 14
-        // Has an if statement depending on the sign of measurement [047]
-        double stepFourteenY = 0.0;
-        if (r_CrotchZ < 0)
-        {
-            stepFourteenY = ((v_SeatBkZ - r_CrotchZ) - (2.5 * Arb_CrotchReduction));
-        }
-        else if (r_CrotchZ > 0)
-        {
-            stepFourteenY = ((v_SeatBkZ + r_CrotchZ) - (2.5 * Arb_CrotchReduction));
-        }
-        else if (r_CrotchZ == 0)
-        {
-            stepFourteenY = ((v_SeatBkZ) - (2.5 * Arb_CrotchReduction));
-        }
+        double centreXPosition = 0;
+        double centreYPosition = 0;
 
-        fullBlock.addKeypoint(new Vector2D((-(k_BodyRise - j_WaistToSeat)), stepFourteenY));
+        double crotchXPosition = centreXPosition;
+        double kneeXPosition = centreXPosition + n_CrotchHeight-m_KneeCRHeight;
+        double ankleXPosition = n_CrotchHeight- l_AnkleCRHeight;
+        double hipXPosition = -(o_HipCHeight-n_CrotchHeight);
+        double seatXPosition = -(p_SeatCHeight-n_CrotchHeight);
+        double waistXPosition = -k_BodyRise;
 
-        // Adding Step 15
-        // Has an if statement depending on the sign of measurement [047]
-        double stepFifteenY = 0.0;
-        if (r_CrotchZ < 0)
-        {
-            stepFifteenY = ((t_HipBkZ - r_CrotchZ) - (2.5 * Arb_CrotchReduction));
-        }
-        else if (r_CrotchZ > 0)
-        {
-            stepFifteenY = ((t_HipBkZ + r_CrotchZ) - (2.5 * Arb_CrotchReduction));
-        }
-        else if (r_CrotchZ == 0)
-        {
-            stepFifteenY = ((t_HipBkZ) - (2.5 * Arb_CrotchReduction));
-        }
+        double halfWaistSuppression = ((h_FrHipArc+i_BkHipArc)/2)-((e_FrWaistArc+f_BkWaistArc)/2);
 
-        fullBlock.addKeypoint(new Vector2D((-(k_BodyRise - a_WaistToHip)), stepFifteenY));
+        ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+        /* Back half points, curves and dart */
+        ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-        // Adding Step 16
-        fullBlock.addKeypoint(new Vector2D(Arb_CrotchPointFive, 0.0));
+        //pre-calculation of point 7 y coordinate - used in a lot of calculations
```

Comment:
Please leave a space between a comment // and the first word and also capitalise it. Otherwise, comments can easily be confused with commented out code.